homeassistant/components/modbus/__init__.py

```diff
@@ -227,13 +233,13 @@
                 {
                     CONF_ADDRESS: cv.positive_int,
                     CONF_HVAC_MODE_VALUES: {
-                        vol.Optional(HVACMode.OFF.value): cv.positive_int,
-                        vol.Optional(HVACMode.HEAT.value): cv.positive_int,
-                        vol.Optional(HVACMode.COOL.value): cv.positive_int,
-                        vol.Optional(HVACMode.HEAT_COOL.value): cv.positive_int,
-                        vol.Optional(HVACMode.AUTO.value): cv.positive_int,
-                        vol.Optional(HVACMode.DRY.value): cv.positive_int,
-                        vol.Optional(HVACMode.FAN_ONLY.value): cv.positive_int,
+                        vol.Optional(CONF_HVAC_MODE_OFF): cv.positive_int,
+                        vol.Optional(CONF_HVAC_MODE_HEAT): cv.positive_int,
+                        vol.Optional(CONF_HVAC_MODE_COOL): cv.positive_int,
+                        vol.Optional(CONF_HVAC_MODE_HEAT_COOL): cv.positive_int,
+                        vol.Optional(CONF_HVAC_MODE_AUTO): cv.positive_int,
+                        vol.Optional(CONF_HVAC_MODE_DRY): cv.positive_int,
+                        vol.Optional(CONF_HVAC_MODE_FAN_ONLY): cv.positive_int,
                     },
```

Comment:
This is a breaking change. Is that really needed?

IMHO we should avoid breaking changes when possible, and I think it is perfectly possible in this case.

This PR should therefore be reverted or adjusted.

CC @janiversen

Reply:
yes it is needed....apart from that the "old" mode is only 1-2 weeks old.

With this change the configuration of modbus is a lot more consistent, and that too is important.

Reply:
@frenck In reality I should have caught this in the previous commit, but overlooked it due to stress.

Reply:
Alright clear, understood. 👍